### PR TITLE
Improve UPCEAN to closer match the Java library and improve accuracy of UPCEAN tests

### PR DESCRIPTION
**Pull request by Swiftwork**
* Fixed unused `resultString` in `UPCEANExtension2Support` and `UPCEANExtension5Support`
* Usage of `StringBuilder` more widespread
* Overloading methods

### CODE REVIEW

**Comment by odahcam** [comment]
What pattern did you adopted for "overloading methods"?

File: src/core/oned/UPCAReader.ts [hunk 1]
```diff
@@ -58,13 +55,16 @@ export default class UPCAReader extends UPCEANReader {
...
 
-  // @Override
-  public decodeRow(rowNumber: number, row: BitArray, hints?: Map<DecodeHintType, any>): Result {
-    return this.maybeReturnResult(this.ean13Reader.decodeRow(rowNumber, row, hints));
+  public decodeRow(rowNumber: number, row: BitArray, hints?: Map<DecodeHintType, any>): Result;
+  public decodeRow(rowNumber: number, row: BitArray, startGuardRange: Int32Array, hints?: Map<DecodeHintType, any>): Result;
+  public decodeRow(rowNumber: number, row: BitArray, arg3: Int32Array | Map<DecodeHintType, any>, arg4?: Map<DecodeHintType, any>): Result {
+    const startGuardRange = arg3 instanceof Int32Array ? arg3 : UPCEANReader.findStartGuardPattern(row);
+    const hints = arg3 instanceof Map ? arg3 : arg4;
+    return this.maybeReturnResult(this.ean13Reader.decodeRow(rowNumber, row, startGuardRange, hints));
```
Comment (odahcam) [inline src/core/oned/UPCAReader.ts:63]: Actually I saw this for overloading, I think we could improve this with some other ideas to still keep the code of each overload close to the Java version (that would be possible with multiple methods with different names but original code inside) and at the same time use only one method with the overload's name (like decodeRow) that would act as an entry point where it's implementation would choose between the implementations.

**Review by Swiftwork — COMMENTED** [review]
Do you prefer this pattern @odahcam

File: src/core/oned/AbstractUPCEANReader.ts [hunk 2]
```diff
@@ -195,7 +195,10 @@ export default abstract class AbstractUPCEANReader extends OneDReader {
   static findGuardPattern(row: BitArray, rowOffset: number, whiteFirst: boolean, pattern: Int32Array, counters: Int32Array): Int32Array;
   static findGuardPattern(row: BitArray, rowOffset: number, whiteFirst: boolean, pattern: Int32Array, counters?: Int32Array): Int32Array {
     if (typeof counters === 'undefined') counters = new Int32Array(pattern.length);
+    return this.findGuardPatternOverload(row, rowOffset, whiteFirst, pattern, counters);
+  }
 
+  static findGuardPatternOverload(row: BitArray, rowOffset: number, whiteFirst: boolean, pattern: Int32Array, counters: Int32Array): Int32Array {
```
Comment (odahcam) [inline src/core/oned/AbstractUPCEANReader.ts:201]: Now that I saw this it got me thinking: should we adopt something like `<methodName>Impl()` like some other languages do? That makes sense?

**Review by odahcam — APPROVED** [review]

File: src/core/oned/AbstractUPCEANReader.ts (hunk 2 above)
Comment (Swiftwork) [reply src/core/oned/AbstractUPCEANReader.ts:201]: Kinda what I had first with `findGuardPatternImplementation`, but I am happy with `findGuardPatternImpl`. I will merge this after the change.

Comment (odahcam) [reply src/core/oned/AbstractUPCEANReader.ts]: Yeah I chose the `<methodName>_Overload[1-9]` in a hurry, don't think that that was a good approach, I should've gone with the `Implementation` word from the start, but `Impl` seems good.

**Pull request merged**